packages/atomic/src/components/search/atomic-tab-manager/atomic-tab-manager.spec.ts

```diff
@@ -0,0 +1,140 @@
+            <atomic-tab label="Articles" name="articles"></atomic-tab>
+          `
+          }
+        </atomic-tab-manager>
+      `,
+      selector: 'atomic-tab-manager',
+    });
+
+    return {element, fakeTabManager};
+  };
+
+  it('should be defined', () => {
+    const el = document.createElement('atomic-tab-manager');
+    expect(el).toBeInstanceOf(HTMLElement);
+  });
+
+  describe('when rendering with valid props', () => {
+    it('should render successfully', async () => {
+      const {element} = await renderTabManager();
+      expect(element).toBeDefined();
+    });
+
+    it('should have tab area part', async () => {
+      const {element} = await renderTabManager();
+      const tabArea = element.shadowRoot?.querySelector('[part="tab-area"]');
+      expect(tabArea).toBeDefined();
+    });
+
+    it('should render tab bar', async () => {
+      const {element} = await renderTabManager();
+      const tabBar = element.shadowRoot?.querySelector('atomic-tab-bar');
+      expect(tabBar).toBeDefined();
+    });
+  });
+
+  describe('when initializing', () => {
+    it('should call buildTabManager with the engine', async () => {
+      const {element} = await renderTabManager();
+
+      expect(buildTabManager).toHaveBeenCalledWith(element.bindings.engine);
+    });
+  });
+
+  describe('when rendering tabs', () => {
+    it('should have the correct number of tabs', async () => {
+      const {element} = await renderTabManager();
+      const tabElements = element.querySelectorAll('atomic-tab');
+      expect(tabElements.length).toBe(3);
+    });
+  });
+
+  describe('when no atomic-tab children exist', () => {
+    let consoleErrorSpy: ReturnType<typeof vi.spyOn>;
+
+    beforeEach(() => {
+      consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(() => {});
+    });
+
+    it('should set error when no atomic-tab children', async () => {
+      const {element} = await renderTabManager({
+        slottedContent: html``,
+      });
+
+      expect(element.error).toBeDefined();
+      expect(element.error.message).toBe(
+        'The "atomic-tab-manager" element requires at least one "atomic-tab" child.'
+      );
+
+      consoleErrorSpy.mockRestore();
+    });
+  });
+
+  describe('when atomic-tab child is missing name attribute', () => {
+    let consoleErrorSpy: ReturnType<typeof vi.spyOn>;
+
+    beforeEach(() => {
+      consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(() => {});
+    });
+
+    it('should set error when atomic-tab is missing name', async () => {
+      const {element} = await renderTabManager({
+        slottedContent: html`
+          <atomic-tab label="Test"></atomic-tab>
+        `,
+      });
+
+      expect(element.error).toBeDefined();
+      expect(element.error.message).toBe(
+        'The "name" attribute must be defined on all "atomic-tab" children.'
+      );
+
+      consoleErrorSpy.mockRestore();
+    });
+  });
```

Comment:
We should add a few more tests I think:

- should call tab select when clicking inactive tab
- should localize tab labels using i18n.t [or something of the sort]
- should pass clearFiltersOnTabChange to buildTab options